StripeFinancialConnections/StripeFinancialConnections/Source/Native/NetworkingLinkLoginWarmup/NetworkingLinkLoginWarmupDataSource.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  NetworkingLinkLoginWarmupDataSource.swift
+//  StripeFinancialConnections
+//
+//  Created by Krisjanis Gaidis on 2/6/23.
+//
+
+import Foundation
+@_spi(STP) import StripeCore
+
+protocol NetworkingLinkLoginWarmupDataSource: AnyObject {
+    var manifest: FinancialConnectionsSessionManifest { get }
+    var analyticsClient: FinancialConnectionsAnalyticsClient { get }
+
+    func disableNetworking() -> Future<Void>
+}
+
+final class NetworkingLinkLoginWarmupDataSourceImplementation: NetworkingLinkLoginWarmupDataSource {
+
+    let manifest: FinancialConnectionsSessionManifest
+    private let apiClient: FinancialConnectionsAPIClient
+    private let clientSecret: String
+    let analyticsClient: FinancialConnectionsAnalyticsClient
+
+    init(
+        manifest: FinancialConnectionsSessionManifest,
+        apiClient: FinancialConnectionsAPIClient,
+        clientSecret: String,
+        analyticsClient: FinancialConnectionsAnalyticsClient
+    ) {
+        self.manifest = manifest
+        self.apiClient = apiClient
+        self.clientSecret = clientSecret
+        self.analyticsClient = analyticsClient
+    }
+
+    func disableNetworking() -> Future<Void> {
+        return apiClient.disableNetworking(
+            disabledReason: nil,
```

Comment:
would we pass a different reason in some cases?

Reply:
@vardges-stripe yes there is [one case](https://git.corp.stripe.com/stripe-internal/stripe-js-v3/blob/10f7ffc3/src/linkedAccounts/inner/components/panes/LinkLogin/useSaveToNetworkAndLink.ts#L124) in Stripe.js, but this particular case does not have a disabled reason 

(note that the strict following of Stripe.js [and the avoidance of extra in-depth research] is intentional for shipping speed)